tests/strategies/test_collaborative.py

```diff
@@ -0,0 +1,344 @@
+            barrier.wait()
+
+    model = TestModel()
+    trainer = pl.Trainer(
+        max_epochs=1,
+        limit_train_batches=16,
+        limit_val_batches=0,
+        strategy=CollaborativeStrategy(
+            delay_state_averaging=True,
+            offload_optimizer=True,
+            delay_optimizer_step=True,
+            delay_grad_averaging=True,
+            target_batch_size=8,
+            initial_peers=initial_peers,
+            verbose=False,
+        ),
+    )
+    trainer.fit(model)
+
+    recorded_process_peers.append(recorded_peers)
+    recorded_process_steps.append(recorded_global_steps)
+
+
+@RunIf(hivemind=True)
+@mock.patch.dict(os.environ, {"HIVEMIND_MEMORY_SHARING_STRATEGY": "file_descriptor"}, clear=True)
+@pytest.mark.parametrize(
+    "num_processes, wait_seconds",
+    [(2, 0.25)],
+)
+def test_multiple_peers(num_processes, wait_seconds):
+    """Test to ensure that if we have two running processes with the same peers, they connect and train
+    successfully."""
+    dht_root = hivemind.DHT(start=True)
+    barrier = mp.Barrier(num_processes)
+    initial_peers = dht_root.get_visible_maddrs()
+
+    with mp.Manager() as manager:
+        # allows processes to return their recorded logged peers/steps
+        recorded_process_peers = manager.list()
+        recorded_process_steps = manager.list()
+        processes = [
+            mp.Process(
+                target=_run_collab_training_fn,
+                kwargs=dict(
+                    initial_peers=initial_peers,
+                    wait_seconds=wait_seconds,
+                    barrier=barrier,
+                    recorded_process_peers=recorded_process_peers,
+                    recorded_process_steps=recorded_process_steps,
+                ),
+            )
+            for x in range(num_processes)
+        ]
+        for process in processes:
+            process.start()
+        for process in processes:
+            process.join()
+        # assert that peers increase as expected and we run at-least 1 global step.
+        for process_peers, process_steps in zip(recorded_process_peers, recorded_process_steps):
+            assert any(num_peer == num_processes for num_peer in process_peers)
+            assert any(global_step > 0 for global_step in process_steps)
+
+
+@RunIf(hivemind=True)
+@mock.patch("torch.cuda.is_available", return_value=True)
+@mock.patch("torch.cuda.device_count", return_value=1)
+@mock.patch.dict(os.environ, {"HIVEMIND_MEMORY_SHARING_STRATEGY": "file_descriptor"}, clear=True)
+def test_scaler_updated_precision_16(mocked_cuda_available, mocked_device_count):
+    class TestModel(BoringModel):
+        def on_fit_start(self) -> None:
+            assert isinstance(self.trainer.precision_plugin.scaler, hivemind.GradScaler)
+            raise SystemExit
+
+    model = TestModel()
+    trainer = pl.Trainer(
+        strategy=CollaborativeStrategy(target_batch_size=1),
```

Comment:
I think it's the first strategy in PyTorch Lightning without an alias 🦜

Reply:
I was thinking about how I could make an alias, as it would be neat to do it from the command line, but I felt like for now we shouldn't try to shoehorn this in

From my experiments, you're always having to extend the class with a bunch of parameters! I think once we start making more QoL improvements we can revisit